src/compiler/utilities.ts

```diff
@@ -2293,12 +2293,12 @@ namespace ts {
             if (options.jsx === JsxEmit.Preserve) {
                 if (isSourceFileJavaScript(sourceFile)) {
                     if (fileExtensionIs(sourceFile.fileName, ".jsx")) {
-                        extension = ".jsx";
+                        extension = options.jsxExtName || ".jsx";

```

Comment:
This means that the user has to remember to add the dot in the extension name. I'm not necessarily sure whether that's desirable.